rules/__tests__/require-tothrow-message.js

```diff
@@ -7,23 +7,44 @@ const ruleTester = new RuleTester();
 
 ruleTester.run('require-tothrow-message', rule, {
   valid: [
-    "expect(function() { a() }).toThrow('a');",
-    "expect(function() { a() }).toThrowError('a');",
+    // String
+    "expect(function() { throw new Error('a'); }).toThrow('a');",
+    "expect(function() { throw new Error('a'); }).toThrowError('a');",
+
+    // Template literal
+    // {
+    //   code:
+    //     "const a = 'a'; expect(() => { throw new Error('a'); })" +
+    //     '.toThrow(`${a}`);',
+    //   parser: 'babel-eslint',
```

Comment:
I think it should work if you set `ecmaVersion`: https://eslint.org/docs/user-guide/configuring#specifying-parser-options